src/index.ts

```diff
@@ -1,16 +1,50 @@
+import { Fraction, transitiveOrderbook } from '@gnosis.pm/dex-contracts'
 import express from 'express'
+import Web3 from 'web3'
+import { CategoryServiceFactory, CategoryConfiguration, Category, LogLevel } from 'typescript-logging'
+import { OrderbookFetcher } from './orderbook_fetcher'
+import * as yargs from 'yargs'
+
+const argv = yargs
+  .option('node', {
+    describe: 'RPC endpoint to connect to',
+    demand: true,
+  })
+  .option('pageSize', {
+    describe: 'The number of orders to fetch per page',
+    default: 50,
+  })
+  .option('port', {
+    describe: 'Port to bind on',
+    default: '8080',
+  })
+  .option('verbosity', {
+    describe: 'log level',
+    choices: ['TRACE', 'DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'],
+    default: 'INFO',
+  }).argv
+
 const app = express()
-const port = 8080 // default port to listen
+const web3 = new Web3(argv.node as string)
+
+CategoryServiceFactory.setDefaultConfiguration(new CategoryConfiguration(LogLevel.fromString(argv.verbosity)))
+const logger = CategoryServiceFactory.getLogger(new Category('dex-price-estimation'))
+const orderbooksFetcher = new OrderbookFetcher(web3, argv.pageSize, logger)
 
 /* tslint:disable:no-unused-expression */
 
-// define a route handler for the default home page
-app.get('/', (_, res) => {
-  res.send('Hello world!')
+app.get('/orderbook', (req, res) => {
+  const transitive = transitiveOrderbook(orderbooksFetcher.orderbooks, req.query.base, req.query.quote, 2)
+  res.json(JSON.stringify(transitive))
+})
+
+app.get('/price', (req, res) => {
+  const transitive = transitiveOrderbook(orderbooksFetcher.orderbooks, req.query.base, req.query.quote, 2)
```

Comment:
nit: I wouldn't add any logic here other than getting the params from the request, and returning the result. The logic is better separated in a function exposed somewhere else.